tensorflow_io/core/plugins/s3/s3_filesystem.cc

```diff
@@ -62,29 +62,61 @@ constexpr size_t kUploadRetries = 3;
 
 constexpr size_t kS3ReadAppendableFileBufferSize = 1024 * 1024;  // 1 MB
 
-static void* plugin_memory_allocate(size_t size) { return calloc(1, size); }
-static void plugin_memory_free(void* ptr) { free(ptr); }
+static inline char* TF_MakeStatusMessage(const Aws::String& str) {
+  char* msg = new char[str.length() + 1];
+  strcpy(msg, str.c_str());
+  return msg;
+}
 
 static inline void TF_SetStatusFromAWSError(
     const Aws::Client::AWSError<Aws::S3::S3Errors>& error, TF_Status* status) {
-  switch (error.GetResponseCode()) {
-    case Aws::Http::HttpResponseCode::FORBIDDEN:
-      TF_SetStatus(status, TF_FAILED_PRECONDITION,
-                   "AWS Credentials have not been set properly. "
-                   "Unable to access the specified S3 location");
-      break;
-    case Aws::Http::HttpResponseCode::REQUESTED_RANGE_NOT_SATISFIABLE:
-      TF_SetStatus(status, TF_OUT_OF_RANGE, "Read less bytes than requested");
-      break;
```

Comment:
I am not sure if this is correct or not since the old code already supported `REQUESTED_RANGE_NOT_SATISFIABLE = 416`. WDYT @yongtang ?
https://github.com/tensorflow/io/blob/9b5e23391dd5fce214ca68fad1ad345e52ce3cac/tensorflow_io/core/plugins/s3/s3_filesystem.cc#L507-L512

Reply:
@vnvo2409 The switch is controlled in:
https://github.com/tensorflow/io/blob/9b5e23391dd5fce214ca68fad1ad345e52ce3cac/tensorflow_io/core/plugins/s3/s3_filesystem.cc#L519

You can change the default value to `true` and see if the issue is resolved.

Reply:
I am investigating this issue https://github.com/tensorflow/tensorflow/issues/44564. Let's leave it for another PR.

Reply:
@yongtang

https://github.com/tensorflow/tensorflow/issues/44564 is resolved. We could enable this option